### DISCUSSION TITLE
Use repository name/pattern within package rule

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#23017] [Suggest an Idea]
### Tell us more.

We have a convention of suffixing any old/inactive git repositories with `-archived`.

I'm interested in being able to define a packageRule that matches the format of the repository name.

This rule would tell Renovate to close all previously opened Renovate PRs.

**Comment by rarkins** [comment]
Would this work?

name: `matchRepository`

Array of strings

Can either be an exact match (e.g. "foo/bar"), or our regex pattern (e.g. "/-archived$"/)

**Reply by setchy** [reply]
Yep, `matchRepository` would work.

I've currently got a workaround that 

1.  runs an auto-discovery via Renovate and saves the discovered repos to JSON
2. intermediate step that parses the JSON output and keeps those matching `-archived$`
3. sets a specialized Renovate pipeline which globally sets several `enabled: "false"` rules to override any repository/shared config

Something like `matchRepository` would help simplify steps 1. and 2 and allow our standard pipeline to execute and leverage an `archived` shared local preset

**Reply by setchy** [reply]
#23085
